src/inmanta/server/services/databaseservice.py

```diff
@@ -48,6 +48,11 @@ async def start(self) -> None:
             self.schedule(
                 self._purge_agent_processes, interval=agent_process_purge_interval, initial_delay=0, cancel_on_stop=False
             )
+        # Schedule database pool exhaustion watch
+        interval: int = 60 * 5  # Check for pool exhaustion every 5 min
```

Comment:
The amount of connections being used varies continuously. If we only check this once every five minutes, it might take a very long time until we notice that we have connection pool issues. I checked with Wouter and he proposed to run the check five times per second because the check itself  is very light weight.